analyzers/ImpactParameter.py

```diff
@@ -1,48 +1,214 @@
 from heppy.framework.analyzer import Analyzer
-from ROOT import TVector3
-#from heppy.framework.event import Event
+from ROOT import TFile, TH1F
+from ROOT import TVector3, TLorentzVector
 from heppy.papas.path import Helix
-#import collections
+import math
+from heppy.utils.deltar import deltaR
 
 class ImpactParameter(Analyzer):
-
+    '''This analyzer puts an impact parameter for every charged particle
+    as an attribute of its path.
+    The significance is calculated, the calculus are a first order approximation,
+    thus this may not be correct for large impact parameters (more than 3 mm).
+    The Impact parameter significance is stored in the particle's path.
+    
+    New attributes for particles.path (from compute_IP) :
+    *   closest_t = time of closest approach to the primary vertex.
+    *   IP = signed impact parameter
+    *   IPx, IPy, IPz = coordinates of the point of closest approach to the
+        primary vertex, with respets to the primary vertex
+        
+    New attributes for particles.path (from compute_theta_0) :
+    *   theta_0 = 1/sqrt(2) * gaussian width of the scattering angle
+        due to the beam pipe
+    *   xX_0 = distance in number of X_0 the radiation length the particles goes through
+    
+    New attributes for particles.path (from compute_IP_signif) :
+    *   IP_signif = the impact parameter significance. To get the uncertainty, just compute IP/IP_signif
         
+    Then, several b-tagging methods are applied to each jet, with selected tracks :
+    *   a log-likelihood ratio based on Impact Parameter (IP_b_LL),
+        if a numerator and a denominator for IP are provided (see example)
+    *   a log-likelihood ratio based on Impact Parameter Significance (IPs_b_LL),
+        if a numerator and a denominator for IPs are provided (see example)
+    *   TCHE and TCHP taggers : using the second and third highest Impact Prameter for each jet
+    
+    New attributes for jets :
+    *   TCHE = the value of the TCHE for this jet
+    *   TCHP = the value of the TCHP for this jet
+    *   TCHE_IP = the value of the IP of the track used to compute the TCHE
+    *   TCHP_IP = the value of the IP of the track used to compute the TCHE
+    *   TCHE_x = the x position of the vertex for the particle used for the TCHE
+    *   TCHE_y = the y position of the vertex for the particle used for the TCHE
+    *   TCHE_z = the z position of the vertex for the particle used for the TCHE
+    *   TCHE_pt = the transverse impulsion of the particle used for the TCHE
+    *   TCHE_dr = the cone containing the track used for the THCE with respects to the jet axis
+    
+    Example of configuration : the root files contain the distributions of IP/IPs histograms (h_u, h_b ...)
```

Comment:
of IP/IPs -> of IP or IPs 
otherwise feels like a ratio (and we have LL ratios so this is misleading)